test/build.test.ts

```diff
@@ -0,0 +1,51 @@
+import { describe, it, expect } from "vitest";
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import { build } from "../src/index.js";
+
+type BuildConfig = Parameters<typeof build>[2];
+
+describe("Build fixtures", () => {
+  const tests: [string, BuildConfig & { dir: string }][] = [
+    [
+      "declaration (mkdist)",
+      {
+        dir: "fixtures/mkdist-declaration",
+      },
+    ],
+    [
+      "sourcemap (mkdist)",
+      {
+        dir: "fixtures/mkdist-sourcemap",
+      },
+    ],
+    [
+      "declaration (rollup)",
+      {
+        dir: "fixtures/rollup-declaration",
+      },
+    ],
+    [
+      "sourcemap (rollup)",
+      {
+        dir: "fixtures/rollup-sourcemap",
+      },
+    ],
+  ];
+
+  it.each(tests)("%s", async (_, { dir, ...buildOptions }) => {
+    const cwd = new URL(dir.replace(/\/?$/, "/"), import.meta.url);
+    await build(fileURLToPath(cwd), false, buildOptions);
```

Comment:
Note: the tests are set up to allow any build options if needed, even though this currently doesn’t take use of that. But we could test stubs, etc. if needed! (Should we add more tests for those?)